src/core/reducer.test.ts

```diff
@@ -749,3 +756,36 @@ describe('undo / redo with stages', () => {
     expect(state.ctx.activePlayers['0']).toBe('B');
   });
 });
+
+describe('TransientHandlingMiddleware', () => {
```

Comment:
Related to returning the result of `dispatch`, I don’t think these tests are testing what you expect. The `result` here is actually just the action object, so for example:

```js
const result = store.dispatch(makeMove('A'));
// result = {
//   type: 'MAKE_MOVE',
//   payload: {
//     type: 'A',
//     args: undefined,
//     playerID: undefined,
//     credentials: undefined,
//   },
// };
```

Because the tests assert absence of the transient keys, they pass because those keys are also not present in the action object.

Instead, you need to get the state from the store after dispatching:

```js
store.dispatch(makeMove('A'));
const result = store.getState();
```